runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -594,26 +595,45 @@ public void closeTable(String namespace, String tableName) {
     }
 
     /**
-     * Only clears a table, DOES not delete its file descriptors from metadata.
-     * This is because if the purpose of the delete is to upgrade from an old schema to a new schema
-     * Then we must first purge all SMR entries of the current (old) format from corfu stream.
-     * It is the checkpointer which actually purges via a trim operation.
-     * But to even get to the trim, it must be able to read the table, which it can't do if the table's
-     * metadata (TableRegistry entry) is also removed here.
-     * So what would be nice if we can place a marker indicating that this table is marked
-     * for deletion, so that on the next Re-open we will actually force update the table registry entry.
-     * For now, just force the openTable to always purge the prior entry assuming that the
-     * caller has done the good work of
-     * 1. table.clear() using the old version
-     * 2. running checkpoint and trim
-     * 3. re-open the table with new proto files.
+     * Special method to remove the table metadata from registry without actually touching
+     * the table itself.
      *
      * @param namespace Namespace of the table.
      * @param tableName Name of the table.
+     * @return number of protobuf files also deleted as a result of this table deletion
      */
-    public void deleteTable(String namespace, String tableName) {
-        Table<Message, Message, Message> table = getTable(namespace, tableName);
-        table.clearAll();
+    public int deleteTable(String namespace, String tableName) {
```

Comment:
shouldnt we keep the old deleteTable() and introduce the new behavior in a separate api?

Reply:
the old deleteTable() not wiping out metadata was the primary issue this PR is trying to fix.

Reply:
ok.. so now we will not have an api in TableRegistry which just deletes the entries.  To do so, clearAll() from Table must be used.  Is that right?

Reply:
clearAll() is a dangerous method to invoke if the table is corrupted. It basically attempts to first materialize the table in-memory which itself can fail if the table is too large. what we need instead is to completely side-step the table so we don't get hit by any serialization/largetable/ memory/disk issues while trying to open a bad table.